test/ui/BreadcrumbValuesListTest.ts

```diff
@@ -0,0 +1,113 @@
+      const title = $$(built).find('.coveo-facet-breadcrumb-title');
+      expect($$(title).text()).toContain('My facet title');
+    });
+
+    it('should display a clickable label for each selected values', () => {
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const values = $$(built).findAll('.coveo-facet-breadcrumb-value');
+      expect(values.length).toBe(3);
+    });
+
+    it('should display a caption for each selected values', () => {
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const captions = $$(built).findAll('.coveo-facet-breadcrumb-caption');
+      expect($$(captions[0]).text()).toContain('foo0');
+      expect($$(captions[1]).text()).toContain('foo1');
+      expect($$(captions[2]).text()).toContain('foo2');
+    });
+
+    it('should display a clear button for each selected values', () => {
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const clear = $$(built).findAll('.coveo-facet-breadcrumb-clear');
+      expect(clear.length).toBe(3);
+    });
+
+    it('should display excluded values differently', () => {
+      facetValues[2].excluded = true;
+      facetValues[2].selected = false;
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const excluded = $$(built).findAll('.coveo-facet-breadcrumb-value.coveo-excluded');
+      expect(excluded.length).toBe(1);
+    });
+
+    it('should respect the facet option for the number of values in breadcrumb', () => {
+      facet.options.numberOfValuesInBreadcrumb = 2;
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const andShowMore = $$(built).findAll('.coveo-facet-breadcrumb-multi-count');
+      expect(andShowMore.length).toBe(1);
+      expect($$(andShowMore[0]).text()).toContain('1 more...');
+    });
+
+    it('should add the excluded and selected values together when it exceeds the number of values in breadcrumb', () => {
+      facet.options.numberOfValuesInBreadcrumb = 1;
+      facetValues[2].excluded = true;
+      facetValues[2].selected = false;
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const andShowMore = $$(built).findAll('.coveo-facet-breadcrumb-multi-count');
+      expect(andShowMore.length).toBe(1);
+      expect($$(andShowMore[0]).text()).toContain('2 more...');
+    });
+
+    it('should show hidden values on clicking show more', () => {
+      facet.options.numberOfValuesInBreadcrumb = 1;
+      facetValues[2].excluded = true;
```

Comment:
Are there other value combinations of `.excluded()` and `.selected()` that are worth testing? Or the same combination but on two different facet values?

Reply:
I could add the test for the edge case where everything is excluded and everything included. 

We're already testing for "some are include and some are excluded", which I think is good enough.

I am not sure what you mean by > same combination but on two different facet values?

Reply:
I was wondering if there may cases worth testing with facetValue[2] set to `excluded=true` and facetValue[3] set as `selected=false` as opposed to setting both properties on facetValue[2]?

Reply:
Ahhh I see.

Humm testing for edge case make sense and should be done (because those tends to produce bugs), but adding too much test for those kind of small "minor" variation can become kind of annoying/time consuming for not much gain in term of issue prevented, in my opinion.